justfile

```diff
@@ -1,23 +1,60 @@
-alias r := run
+# Use values from .env automatically, if present
+set dotenv-load
 
+alias r := run
+alias b  := build
+alias bb := bbuild
 
-# platform := "--platform=linux/amd64"
+# example: platform := "--platform=linux/amd64"
 platform := ""
 image := "rnacentral/r2dt"
 base_image := "rnacentral/r2dt-base"
+data_dir := "./1.4"
 
 
 # Default recipe to display help information
 default:
   @just --list
 
+# Prepare and activate Python virtual environment
+venv:
+    python3 -m venv .venv
+    pip3 install -r requirements.txt
+    source .venv/bin/activate
+
+# Download data from RNAcentral
+download:
+    curl -O https://ftp.ebi.ac.uk/pub/databases/RNAcentral/r2dt/1.4/cms.tar.gz
+    tar -xzf cms.tar.gz
+
 # Run shell in docker
 run:
-  docker run {{platform}} -v $(pwd):/rna/r2dt -v ./1.4:/rna/r2dt/data/cms -it --rm {{image}}
+  docker run {{platform}} -v $(pwd):/rna/r2dt -v {{data_dir}}:/rna/r2dt/data/cms -it --rm {{image}}
+
+# Run all tests in docker
+test-all:
+  docker run {{platform}} --rm -it -v ./:/rna/r2dt/ -v {{data_dir}}:/rna/r2dt/data/cms {{image}} bash -c "./r2dt.py test"
+# Run specific test in docker
+test TEST:
+  docker run {{platform}} --rm -it -v ./:/rna/r2dt/ -v {{data_dir}}:/rna/r2dt/data/cms {{image}} bash -c "./r2dt.py test Test{{TEST}}"
+
+# Build R2DT Docker image
+build:
+  docker buildx build {{platform}} -t {{image}}  .
```

Comment:
@anayden Might as well do `docker build` instead of `docker buildx build` as it's an alias anyway - but it's minor of course.

Reply:
Hm…I'm not sure it's always the case. AFAIK, `buildx` is a way to interact with `BuildKit`, which is not always installed. That means `docker build` may work (incorrectly) in case of `BuildKit` is absent, but `docker buildx build` will fail immediately and expose the issue

Reply:
I must have misunderstood [these discussions](https://www.reddit.com/r/docker/comments/14nmv44/docker_builder_variety_is_confusing/) then but I don't have a strong opinion on this one. 🤷

Reply:
> I must have misunderstood [these discussions](https://old.reddit.com/r/docker/comments/14nmv44/docker_builder_variety_is_confusing/)

Hm…Actually, the first comment there says `buildx build` and `build` are almost the same, but `buildx build` is more flexible. Whereas the last one says they are the same. Tbh, I don't know whom to believe :)